diff -uprN a/arch/arm64/boot/dts/rockchip/rk3528-hinlink-ht2.dts b/arch/arm64/boot/dts/rockchip/rk3528-hinlink-ht2.dts
--- a/arch/arm64/boot/dts/rockchip/rk3528-hinlink-ht2.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3528-hinlink-ht2.dts
@@ -169,6 +169,8 @@
 		regulator-always-on;
 		regulator-boot-on;
 		regulator-settling-time-up-us = <250>;
+		regulator-enable-ramp-delay = <400>;
+		regulator-ramp-delay = <12500>;
 		pwm-supply = <&vcc5v0_sys>;
 		status = "okay";
 	};
@@ -183,6 +185,8 @@
 		regulator-always-on;
 		regulator-boot-on;
 		regulator-settling-time-up-us = <250>;
+		regulator-enable-ramp-delay = <400>;
+		regulator-ramp-delay = <12500>;
 		pwm-supply = <&vcc5v0_sys>;
 		status = "okay";
 	};
diff -uprN a/arch/arm64/boot/dts/rockchip/rk3528.dtsi b/arch/arm64/boot/dts/rockchip/rk3528.dtsi
--- a/arch/arm64/boot/dts/rockchip/rk3528.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3528.dtsi
@@ -1017,16 +1017,16 @@
 			     <GIC_SPI 93 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 90 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>;
-		interrupt-names = "Mali_GP_IRQ",
-				  "Mali_GP_MMU_IRQ",
-				  "IRQPP",
-				  "Mali_PP0_IRQ",
-				  "Mali_PP0_MMU_IRQ",
-				  "Mali_PP1_IRQ",
-				  "Mali_PP1_MMU_IRQ";
-		clocks = <&scmi_clk SCMI_CLK_GPU>, <&cru ACLK_GPU_MALI>,
-			 <&cru PCLK_GPU_ROOT>;
-		clock-names = "clk_mali", "aclk_gpu_mali", "pclk_gpu";
+		interrupt-names = "gp",
+				  "gpmmu",
+				  "pp",
+				  "pp0",
+				  "ppmmu0",
+				  "pp1",
+				  "ppmmu1";
+		clocks = <&cru ACLK_GPU_MALI>, <&cru ACLK_GPU_MALI>;
+						 
+		clock-names = "bus", "core";
 		assigned-clocks = <&scmi_clk SCMI_CLK_GPU>;
 		assigned-clock-rates = <300000000>;
 		power-domains = <&power RK3528_PD_GPU>;
